spec/controllers/api/v1/batch_assignment_overrides_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+          expect(assignment.split).to eq split
+          expect(assignment.mixpanel_result).to eq "success"
+          expect(assignment.context).to eq "context"
+          expect(assignment.force).to eq true
+        end
+
+        it "overrides an assignment if one already exists" do
+          FactoryBot.create(:assignment, visitor: visitor, split: split, variant: "control")
+
+          expect {
+            post :create, params: create_params
+          }.to change { PreviousAssignment.count }.by(1)
+
+          expect(Assignment.last.force).to eq true
+          expect(PreviousAssignment.last.force).to eq false
+          expect(response).to have_http_status :no_content
+        end
+
+        it "allows a request without a mixpanel_result" do
+          post :create, params: create_params.except(:mixpanel_result)
+
+          expect(response).to have_http_status :no_content
+
+          assignment = Assignment.first
+          expect(assignment.variant).to eq "treatment"
+          expect(assignment.visitor).to eq visitor
+          expect(assignment.split).to eq split
+          expect(assignment.mixpanel_result).to eq nil
+          expect(assignment.context).to eq "context"
+        end
+
+        context "with multiple assignments in the payload" do
+          let(:split2) { FactoryBot.create(:split, registry: { control: 50, treatment: 50 }) }
+
+          let(:create_params) do
+            {
+              visitor_id: visitor.id,
+              assignments: [
+                {
+                  split_name: split.name,
+                  variant: "treatment",
+                  mixpanel_result: "success",
+                  context: "context"
+                },
+                {
+                  split_name: split2.name,
+                  variant: "treatment",
+                  mixpanel_result: "success",
+                  context: "context"
+                }
+              ]
+            }
+          end
+
+          it "creates an assignment for each" do
+            expect {
+              post :create, params: create_params
+            }.to change { Assignment.count }.by(2)
+
+            expect(response).to have_http_status :no_content
+
+            assignment = Assignment.first
+            expect(assignment.variant).to eq "treatment"
+            expect(assignment.visitor).to eq visitor
+            expect(assignment.split).to eq split
+            expect(assignment.mixpanel_result).to eq "success"
+            expect(assignment.context).to eq "context"
+            expect(assignment.force).to eq true
+
+            assignment = Assignment.second
+            expect(assignment.variant).to eq "treatment"
+            expect(assignment.visitor).to eq visitor
+            expect(assignment.split).to eq split2
+            expect(assignment.mixpanel_result).to eq "success"
+            expect(assignment.context).to eq "context"
+            expect(assignment.force).to eq true
+          end
+        end
+      end
```

Comment:
all the tests in this file _except for this one_ are duplicated from the controller tests for the singular assignment override endpoint. 

this one shows that the behavior is correct when there are multiple inputs